dev/com.ibm.ws.microprofile.reactive.streams.operators_fat/test-applications/ConcurrentWorkTest/src/com/ibm/ws/microprofile/reactive/streams/test/concurrent/ReactiveParallelInServer.java

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.streams.test.concurrent;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.IntStream;
+
+import javax.annotation.Resource;
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+import org.eclipse.microprofile.reactive.streams.operators.ProcessorBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ * This class provides an endpoint for a test that does some
+ * reactive streams FAT tests parallel in the server.
+ */
+@ApplicationScoped
+@Path("/parallelInServer")
+public class ReactiveParallelInServer {
+
+    /**  */
+    private static final int PROC_INCREMENT = 100;
+    /**  */
+    private static final int PIPE_LENGTH = 5;
+    /**  */
+    private static final int FOUNDATION = 1000000;
+    /**  */
+    private static final int ELEMENTS = 10;
+    @Resource
+    private ManagedExecutorService mes; //Injected by the container
+
+    @GET
+    public String restEndpoint(@QueryParam("message") String message) {
+
+        System.out.println(message + " received on thread " + Thread.currentThread().getName());
+
+        // Set up the parallel tasks we create 3 * those that can be run on a physical core
+        int count = 3 * Runtime.getRuntime().availableProcessors();
```

Comment:
is a machine dependent variable really needed for this test? ideally tests should run identically on any machine.